### PR TITLE
feat: web app connect to metamask extension

### PR DESCRIPTION
**Pull request by sidmorizon**

### CODE REVIEW

File: packages/kit/src/components/WalletConnect/useWalletConnectSendInfo.tsx [hunk 1]
```diff
@@ -13,7 +13,7 @@ import {
 } from '@onekeyhq/components';
```
Comment (sidmorizon) [inline packages/kit/src/components/WalletConnect/useWalletConnectSendInfo.tsx:13]: move to standalone externalAccount hooks

File: packages/kit/src/views/Send/components/AuthExternalAccountInfo.tsx [hunk 2]
```diff
@@ -37,10 +37,25 @@ const AuthExternalAccountInfo = React.memo(
       accountInfo,
       client,
       setTitleInfo,
+      injectedConnectorInfo,
     } = props;
     const intl = useIntl();
     const navigation = useAppNavigation();
     const walletName = accountInfo?.walletName;
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    const externalAccountType = accountInfo?.type;
```
Comment (sidmorizon) [inline packages/kit/src/views/Send/components/AuthExternalAccountInfo.tsx:46]: move to standalone externalAccount hooks

File: packages/kit/src/views/Send/utils/useSignOrSendOfExternalAccount.ts [hunk 3]
```diff
@@ -0,0 +1,152 @@
+import { useCallback } from 'react';
+
+import { cloneDeep, isString } from 'lodash';
```
Comment (sidmorizon) [inline packages/kit/src/views/Send/utils/useSignOrSendOfExternalAccount.ts:3]: move to standalone externalAccount hooks

File: packages/kit/src/views/ExternalAccount/SendConfirm/AuthExternalAccountInfo.tsx [hunk 4]
```diff
@@ -37,10 +37,25 @@ const AuthExternalAccountInfo = React.memo(
...
     } = props;
     const intl = useIntl();
     const navigation = useAppNavigation();
     const walletName = accountInfo?.walletName;
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    const externalAccountType = accountInfo?.type;
+    const wcSessionConnected = session?.connected;
+    const { connected } = useMemo(() => {
+      const $connected = injectedConnectorInfo
+        ? // TODO function computeIsActive({ chainId, accounts, activating }) {
```
Comment (sidmorizon) [inline packages/kit/src/views/ExternalAccount/SendConfirm/AuthExternalAccountInfo.tsx:50]: TODO use computeIsActive instead

**Review by sunnylqm — APPROVED** [review]

**Review by revan-zhang — APPROVED** [review]

**Review by kwoktung — APPROVED** [review]

**Pull request merged**